services/localoam.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+var meta = require('../controllers/meta.js');
+var request = require('request');
+
+/**
+* LocalOAM Constructor that handles intractions with local OAM bucket server
+*
+* @constructor
+* @param {String} bucket (optional) - Local OAM Bucket URL. Can be set by LOCAL_OAM_BUCKET env var.
+*/
+var LocalOAM = function (bucket) {
+  this.params = {
+    bucket: bucket || process.env.LOCAL_OAM_BUCKET
+  };
+};
+
+/**
+* Read bucket method for LocalOAM. It reads the LocalOAM bucket and adds all the `.json` metadata to Meta model
+*
+* @param {responseCallback} cb - The callback that handles the response
+* @param {finishedCallback} finished - The callback that handles when reading is done
+*/
+
+LocalOAM.prototype.readBucket = function (lastSystemUpdate, cb, done) {
+  console.info('--- Reading from bucket: ' + this.params.bucket + ' ---');
+
+  var self = this;
+  self.tasks = [];
```

Comment:
Ah, there should only end up being a single `readBucket` per bucket, since each gets its own instance of `S3` or `LocalOAM`.